db/db.pl

```diff
@@ -2267,6 +2345,71 @@ sub parseArgs {
       print "$rmcount user(s) disabled\n";
     }
 
+    exit 0;
+} elsif ($ARGV[1] eq "setshortcut") {
+    showHelp("Invalid name $ARGV[2], names cannot have special characters except '_'") if ($ARGV[2] =~ /[^-a-zA-Z0-9_]$/);
+    showHelp("file '$ARGV[4]' not found") if (! -e $ARGV[4]);
+    showHelp("file '$ARGV[4]' empty") if (-z $ARGV[4]);
+
+    parseArgs(5);
+
+    showHelp("Type must be ip, string, or number instead of $TYPE") if ($TYPE !~ /^(string|ip|number)$/);
+
+    # read shortcuts file
+    my $shortcutValues;
+    open(my $fh, '<', $ARGV[4]);
+    {
+      local $/;
+      $shortcutValues = <$fh>;
+    }
+    close($fh);
+
+    my $shortcutsArray = [split /[\n,]/, $shortcutValues];
+
+    my $shortcutName = $ARGV[2];
+    my $shortcutUserId = $ARGV[3];
+
+    my $shortcuts = esGet("/${PREFIX}lookups/_search?q=name:${shortcutName}");
+
+    my $existingShortcut;
+    foreach my $shortcut (@{$shortcuts->{hits}->{hits}}) {
+      if ($shortcut->{_source}->{name} == $shortcutName) {
+        $existingShortcut = $shortcut;
+        last;
+      }
+    }
+
+    # create shortcut object
+    my $newShortcut;
+    $newShortcut->{name} = $shortcutName;
+    $newShortcut->{userId} = $shortcutUserId;
+    $newShortcut->{$TYPE} = $shortcutsArray;
+    if ($existingShortcut) { # use existing optional fields
+      if ($existingShortcut->{_source}->{description}) {
+        $newShortcut->{description} = $existingShortcut->{_source}->{description};
+      }
+      if ($existingShortcut->{_source}->{shared}) {
+        $newShortcut->{shared} = $existingShortcut->{_source}->{shared};
+      }
+    }
+    if ($DESCRIPTION) {
+      $newShortcut->{description} = $DESCRIPTION;
+    }
+    if ($SHARED) {
+      $newShortcut->{shared} = \1;
```

Comment:
what does \1 even do?  does just 1 work?

Reply:
\1 translates to true. "true" is a string, which we don't want and 1 is a number, which we don't want either
https://stackoverflow.com/questions/43867553/how-to-send-boolean-value-from-perl-script-without-converting-them-into-string